Zend/zend_API.h

```diff
@@ -1223,14 +1223,14 @@ static zend_always_inline zval *zend_try_array_init(zval *zv)
 	_(Z_EXPECTED_DOUBLE_OR_NULL,	"of type ?float") \
 	_(Z_EXPECTED_NUMBER,			"of type int|float") \
 	_(Z_EXPECTED_NUMBER_OR_NULL,	"of type int|float|null") \
-	_(Z_EXPECTED_STRING_OR_ARRAY,	"of type array|string") \
-	_(Z_EXPECTED_STRING_OR_ARRAY_OR_NULL, "of type array|string|null") \
+	_(Z_EXPECTED_ARRAY_OR_STRING,	"of type array|string") \
+	_(Z_EXPECTED_ARRAY_OR_STRING_OR_NULL, "of type array|string|null") \
 	_(Z_EXPECTED_STRING_OR_LONG,	"of type string|int") \
 	_(Z_EXPECTED_STRING_OR_LONG_OR_NULL, "of type string|int|null") \
-	_(Z_EXPECTED_CLASS_NAME_OR_OBJECT,	"a valid class name or object") \
-	_(Z_EXPECTED_CLASS_NAME_OR_OBJECT_OR_NULL, "a valid class name, object, or null") \
-	_(Z_EXPECTED_STRING_OR_OBJECT,	"of type object|string") \
-	_(Z_EXPECTED_STRING_OR_OBJECT_OR_NULL, "of type object|string|null") \
+	_(Z_EXPECTED_OBJECT_OR_CLASS_NAME,	"an object or a valid class name") \
+	_(Z_EXPECTED_OBJECT_OR_CLASS_NAME_OR_NULL, "an object, a valid class name, or null") \
```

Comment:
As "class name" is not a real type, I'm not sure it makes sense to invert it. At least I expect that for this type "class name" will be the common case, not "object".

Reply:
Do you mean the error message only or the C code itself too? I don't really have a strong opinion about the former, but this way, the message sounds better for me. Since class names are strings anyway, I think the current order might have some merit.

Speaking about the C cod itself, `Z_PARAM_CLASS_NAME_OR_OBJ` together with `Z_PARAM_OBJ_OR_STR` feels inconsistent for me. On the other hand, `Z_PARAM_OBJ_OR_CLASS_NAME` is very similar to `Z_PARAM_OBJ_OF_CLASS`.

Do you still have the same preference? If so, I'll change orders.

Reply:
Okay, let's go with this.